### DISCUSSION TITLE
How should I go about making the server 100% bright?

### DISCUSSION THREAD

**Discussion by ryleu** [Minestom/Minestom#1056] [Help] [answered]
I just started using minestom today, and I have a world up and running, but when I join, everything is dark. I asked in the Discord and iam said that the lighting engine is still WIP, so my solution is to just make the whole world fully bright.

How should I go about doing that, if that's the best option? 

Thanks in advance!

**Comment by cody-quinn** [comment, marked as answer]
You can do this by setting the DimensionType to one with an ambientLight value of 2.0
```java
// Creating and registering the dimension
DimensionType fullbright = DimensionType.builder(NamespaceID.from("minestom:full_bright"))
            .ambientLight(2.0f)
            .build();
MinecraftServer.getDimensionTypeManager().addDimension(fullbright);

// Using it to make our instance
MinecraftServer.getInstanceManager.createInstanceContainer(fullbright);
```

**Reply by ryleu** [reply]
what's special about 2.0f? wouldn't 1 be fully bright (light level 15)?

**Reply by cody-quinn** [reply]
> what's special about 2.0f? wouldn't 1 be fully bright (light level 15)?

1.0 might be full bright, I have never tried it. When I ran accross this issue everyone recommended 2.0 and I never bothered to try 1.0 because it worked.

**Reply by ryleu** [reply]
got it. one more question: does it matter which namespace i use? in vanilla, you have to have a datapack and stuff. i assume that's not the case here?

**Reply by cody-quinn** [reply]
You can use any namespace you like

**Reply by ryleu** [reply]
to make this discussion a bit more complete, here's my final code:
```kt
// create the dimension with the namespace `"ryleu:overworld"`
val fullbrightOverworld = DimensionType.builder(NamespaceID.from("ryleu:overworld"))
    .ambientLight(2f)
    .build()

// add the dimension
MinecraftServer.getDimensionTypeManager().addDimension(fullbrightOverworld)

// Create the instance
overworld = instanceManager.createInstanceContainer(fullbrightOverworld)
```

**Comment by MrGazdag** [comment]
You can create a custom [`DimensionType`](https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/world/DimensionType.java) using `DimensionType#builder(NamespaceID)` and you can set the ambient light level using `DimensionTypeBuilder#ambientLight(float)`. To make it fully bright, you can set it to `2f`.

You can apply the custom `DimensionType` to an instance in the instance's constructor, or `InstanceManager.createInstanceContainer(DimensionType)`.
